packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -55,8 +54,8 @@ export interface ComponentDef {
     propsConfig: Record<string, PropType>;
     methods: PropertyDescriptorMap;
     template: Template;
-    preferNativeShadow: boolean;
     renderMode: RenderMode;
+    shadowSupportMode?: 'any';
```

Comment:
I would prefer to make this an enum so we can support multiple values more easily. To avoid conflicting with the existing `ShadowMode` enum (`'native' | 'synthetic'`), maybe it could be `TemplateShadowMode` or something?

Reply:
Yeah, I initially had it as an enum but changed it because there was only one value. I'll change it back.